tls/s2n_connection.c

```diff
@@ -1290,3 +1292,27 @@ uint8_t s2n_connection_get_protocol_version(const struct s2n_connection *conn)
     }
     return conn->server_protocol_version;
 }
+
+int s2n_connection_set_keyshare_by_group_for_testing(struct s2n_connection *conn, uint16_t iana_id)
+{
+    S2N_ERROR_IF(!S2N_IN_TEST, S2N_ERR_NOT_IN_TEST);
```

Comment:
```c
ENSURE_POSIX(S2N_IN_TEST, S2N_ERR_NOT_IN_TEST);
```

Reply:
> ```c
> ENSURE_POSIX(S2N_IN_TEST, S2N_ERR_NOT_IN_TEST);
> ```

We probably should change other places in the code (https://github.com/awslabs/s2n/blob/master/tls/s2n_config.c#L241) with this change.

Reply:
I think it's fine to leave for now, with the idea that any new code should use `ENSURE` instead of `S2N_ERROR_IF`.